op-e2e/e2eutils/disputegame/output_game_helper.go

```diff
@@ -553,6 +559,36 @@ func (g *OutputGameHelper) PreimageExistsInOracle(ctx context.Context, data *typ
 	return exists
 }
 
+func (g *OutputGameHelper) UploadPreimage(ctx context.Context, data *types.PreimageOracleData, privateKey *ecdsa.PrivateKey) {
+	oracle := g.Oracle(ctx)
+	call := oracle.AddGlobalDataTxCall(data)
+	candidate, err := call.ToTxCandidate()
+	g.require.NoError(err, "Failed to create tx candidate")
+
+	logger := testlog.Logger(g.t, log.LvlInfo).New("role", "OutputGameHelper", "game", g.addr)
+	conf := txmgr.NewCLIConfig(g.system.NodeEndpoint("l1"), txmgr.DefaultChallengerFlagValues)
+	conf.NumConfirmations = 1
+	conf.ReceiptQueryInterval = time.Second * 1
+	conf.PrivateKey = e2eutils.EncodePrivKeyToString(privateKey)
+	txMgr, err := txmgr.NewSimpleTxManager("e2etest", logger, metrics.NoopMetrics, conf)
+	g.require.NoError(err, "Failed to create tx manager")
+
+	receipt, err := txMgr.Send(ctx, candidate)
+	g.require.NoError(err, "Failed to send tx")
+	g.require.EqualValues(gethtypes.ReceiptStatusSuccessful, receipt.Status, "Transaction failed")
```

Comment:
Typically the e2e code just uses the generated bindings - largely to avoid needing to create a txmgr here.  If we do create the txmgr we'd also need to Close it again and it all gets a bit ugly.

Reply:
I figured we'll get rid of all bindings usage for faultproofs at some point. This was done to ease that transition. But for now I'll fix this to use the bindings. It's so much nicer than dealing with the txmgr.

Reply:
Yeah I'm in two minds about that.  The new bindings are _so_ much nicer for everything except sending transactions.  But I also like using different implementations between production and test so you don't just get matching bugs in both.